pkg/syncer/metadatasyncer.go

```diff
@@ -398,38 +339,46 @@ func InitMetadataSyncer(ctx context.Context, clusterFlavor cnstypes.CnsClusterFl
 	}
 
 	if metadataSyncer.clusterFlavor == cnstypes.CnsClusterFlavorGuest {
-		// Check the replicated FSS Configmap every 2 minutes
-		// When workload-domain-isolation FSS is enabled in csi-feature-states config-map, restart the container
-		if commonco.ContainerOrchestratorUtility.IsPVCSIFSSEnabled(ctx, common.WorkloadDomainIsolationFSS) &&
-			!commonco.ContainerOrchestratorUtility.IsCNSCSIFSSEnabled(ctx, common.WorkloadDomainIsolationFSS) {
+		// If workload-domain-isolation FSS is not enabled on guest cluster, then check the capabilities CR in
+		// supervisor cluster every 2 mins to check if there is a change in capability value from false to true.
+		// If so, restart the CSI container.
+		// NOTE: We can add other capabilities here when similar functionality is required. For
+		// workload-isolation-domain feature we are restarting the container when capability changes dynamically from
+		// false to true, but for other features instead of restarting CSI container, if possible we can implement
+		// init() function and it can initialize required things when capability value changes from false to true.
+		if !commonco.ContainerOrchestratorUtility.IsPVCSIFSSEnabled(ctx, common.WorkloadDomainIsolationFSS) {
```

Comment:
query - here are we supposed to use IsPVCSIFSSEnabled() or IsFSSEnabled() since IsPVCSIFSSEnabled() checks CSI configmap for FSS and IsFSSEnabled() will check capability CR ??